nat-gateway.tf

```diff
@@ -11,7 +11,9 @@ locals {
   nat_gateway_eip_count   = local.use_existing_eips ? 0 : local.nat_gateways_count
   gateway_eip_allocations = local.use_existing_eips ? data.aws_eip.nat_ips.*.id : aws_eip.default.*.id
   eips_allocations        = local.use_existing_eips ? data.aws_eip.nat_ips.*.id : aws_eip.default.*.id
-  nat_gateways_count      = var.nat_gateway_enabled ? length(var.availability_zones) : 0
+  single_nat_gateway      = var.single_nat_gateway ? 1 : length(var.availability_zones)
```

Comment:
the name of the local `single_nat_gateway` is misleading.

maybe

```
nat_gateways_count_internal      = var.single_nat_gateway ? 1 : length(var.availability_zones)
```

Reply:
this comments